analysis_module/tests/test_apis.py

```diff
@@ -436,3 +443,153 @@ def test_prediction_mock(self, model_prediction_class):
             status=NLPRequest.RequestStatus.SUCCESS,
         ).exists(), "NLP request should not be created for mock request"
         model_prediction_class.assert_not_called()
+
+
+class TestTextExtractionAPI(BaseTestCase):
+    URL = '/api/v1/text-extraction/'
+    CLIENT_ID = "client_id"
+
+    def test_extraction_invalid_data(self):
+        no_documents = {"callback_url": "someurl"}
+        no_callback_url = {"documents": []}
+        no_url = {
+            "documents": [{"client_id": "some client id"}],
+            "callback_url": "some url"
+        }
+        no_client_id = {
+            "documents": [{"url": "someurl"}],
+            "callback_url": "some url"
+        }
+        request_type_string = {
+            "documents": [{"url": "someurl", "client_id": "cid"}],
+            "callback_url": "some url",
+            "request_type": "somerandom",
+        }
+        request_type_invalid = {
+            "documents": [{"url": "someurl", "client_id": "cid"}],
+            "callback_url": "some url",
+            "request_type": 20,  # 20 is not a valid request type
+        }
+        invalid_data = [
+            (no_documents, "documents"),
+            (no_callback_url, "callback_url"),
+            (no_url, "documents"),
+            (no_client_id, "documents"),
+            (request_type_string, "request_type"),
+            (request_type_invalid, "request_type"),
+        ]
+        for params, err_key in invalid_data:
+            self.set_credentials()
+            resp = self.client.post(self.URL, data=params)
+            assert resp.status_code == 400
+            errors = resp.json()["field_errors"]
+            assert err_key in errors
+
+    @patch('analysis_module.utils.requests')
+    def test_extraction_system_request(self, requests_mock):
+        """
+        Http request to ECS will not be called right away
+        """
+        documents = [
+            {"url": "someurl", "client_id": self.CLIENT_ID},
+            {"url": "anothersomeurl", "client_id": self.CLIENT_ID+"1"},
```

Comment:
@bewakes Do we have check for `client_id`? Assume user make request with same client_id for different document url.

Reply:
We don't care about that. It's upto users to maintain uniqueness.